lib/ispmanager_installer.php

```diff
@@ -44,78 +44,110 @@ public function install(stdClass $service, stdClass $meta)
+        $is_lite = $nodeResponse->doc->error->{'$type'} == 'missed';
+
+        $softaculous_url = ($meta->use_ssl == 'true' ? 'https' : 'http') . '://' . $meta->host_name . '/softaculous/';
+        if ($is_lite) {
+            // Make another authentication call in order to set authentication cookies
+            $cookieData = ['func' => 'auth', 'auth' => $session_id, 'sok' => 'ok'];
+            $this->makeRequest($cookieData, $hostUrl, 'GET');
+
+            // Softaculous on ISPmanager requires a CSRF token for each call
+            $configOptions = array_merge($configOptions, $this->getToken($softaculous_url));
+        } else {
+            // Authenticate in to Softaculous
+            $softaculousData = [
+                'auth' => $session_id,
+                'func' => 'softaculous.redirect',
+                'out' => 'json',
+                'sok' => 'ok'
+            ];
+
+            $softaculousResponse = $this->makeRequest(
+                $softaculousData,
+                $hostUrl,
+                'GET'
+            );
+
+            if (!isset($softaculousResponse->doc->ok->{'$'})) {
+                $errorMessage = Language::_('SoftaculousPlugin.remote_error', true);
+                $this->Input->setErrors(['login' => ['invalid' => $errorMessage]]);
+                $this->logger->error(json_encode($softaculousResponse));
+                return;
+            }
+
+            // Get Softaculous login url
+            $login = $softaculousResponse->doc->ok->{'$'};
+            $query = explode('?', $login, 2);
+
+            $login = isset($query[0]) ? $query[0] : null;
+            $query = isset($query[1]) ? $query[1] : null;
+
+            parse_str($query, $query);
+
+            $urlData = [
+                'func' => 'redirect',
+                'auth' => $query['auth'],
+                'authm' => $query['authm'],
+                'lang' => $query['lang'],
+                'redirect_uri' => $query['redirect_uri'],
+                'sok' => 'ok'
+            ];
+
+            $urlResponse = $this->makeRequest(
+                $urlData,
+                $login,
+                'GET'
+            );
+
+            // Softaculous on ISPmanager requires a CSRF token for each call
+            $api = isset($urlResponse->location) ? $urlResponse->location : $login;
+            $configOptions = array_merge($configOptions, $this->getToken($api));
+
+            // Install script
+            $softaculous_url = isset($urlResponse->location) ? $urlResponse->location : $login;
+            // Set installer options
+            $this->setOptions(
+                [
+                    'request' => [
+                        'raw' => false,
+                        'follow_location' => true,
+                        'referer' => $softaculous_url . '?api=serialize&act=software'
+                    ]
                 ]
-            ]
-        );
+            );
```

Comment:
Moving this block of code out of the else block, seems to fix the MySQL error.

Reply:
Updated